homeassistant/components/ovo_energy/__init__.py

```diff
@@ -32,30 +33,35 @@
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up OVO Energy from a config entry."""
 
-    client = OVOEnergy()
+    client = OVOEnergy(
+        client_session=async_get_clientsession(hass),
+    )
 
     try:
-        authenticated = await client.authenticate(
+        if not await client.authenticate(
             entry.data[CONF_USERNAME],
             entry.data[CONF_PASSWORD],
-            entry.data[CONF_ACCOUNT],
-        )
+        ):
+            raise ConfigEntryAuthFailed
+
+        await client.bootstrap_accounts()
+
+        if custom_account := entry.data.get(CONF_ACCOUNT) is not None:
+            client.custom_account_id = custom_account
     except aiohttp.ClientError as exception:
         _LOGGER.warning(exception)
         raise ConfigEntryNotReady from exception
 
-    if not authenticated:
-        raise ConfigEntryAuthFailed
-
     async def async_update_data() -> OVODailyUsage:
         """Fetch data from OVO Energy."""
         async with asyncio.timeout(10):
             try:
                 authenticated = await client.authenticate(
                     entry.data[CONF_USERNAME],
                     entry.data[CONF_PASSWORD],
-                    entry.data[CONF_ACCOUNT],
                 )
+                if custom_account := entry.data.get(CONF_ACCOUNT) is not None:
+                    client.custom_account_id = custom_account
```

Comment:
Please move code out of the try block that is not expected to raise aiohttp.ClientError